gunrock/app/mf/mf_test.cuh

```diff
@@ -0,0 +1,336 @@
+double CPU_Reference(
+    const    GraphT          &graph,
+    // TODO: add problem specific inputs and outputs, e.g.:
+                     ValueT  *excess,
+    typename GraphT::VertexT  src,
+    bool                      quiet)
+{
+#ifdef BOOST_FOUND
+    using namespace boost;
+    typedef typename GraphT::VertexT VertexT;
+    typedef typename GraphT::SizeT   SizeT;
+    typedef typename GraphT::ValueT  GValueT;
+    // TODO: change to other graph representation, if not using Csr
+    typedef typename GraphT::CsrT    CsrT;
+
+    // Prepare Boost Datatype and Data structure
+    typedef adjacency_list<vecS, vecS, directedS, no_property,
+            property <edge_weight_t, GValueT> > BGraphT;
+
+    typedef typename graph_traits<BGraphT>::vertex_descriptor vertex_descriptor;
+    typedef typename graph_traits<BGraphT>::edge_descriptor edge_descriptor;
+
+    typedef std::pair<VertexT, VertexT> EdgeT;
+
+    EdgeT   *edges = ( EdgeT*)malloc(sizeof( EdgeT) * graph.edges);
+    GValueT *weight = (GValueT*)malloc(sizeof(GValueT) * graph.edges);
+
+    for (VertexT v = 0; v < graph.nodes; ++v)
+    {
+        SizeT edge_start = graph.CsrT::GetNeighborListOffset(v);
+        SizeT num_neighbors = graph.CsrT::GetNeighborListLength(v);
+        for (SizeT e = 0; e < num_neighbors; ++e)
+        {
+            edges [e + edge_start] = EdgeT(v, graph.CsrT::GetEdgeDest(e + edge_start));
+            weight[e + edge_start] = graph.CsrT::edge_values[e + edge_start];
+        }
+    }
+
+    BGraphT g(edges, edges + graph.edges, weight, graph.nodes);
+
+    std::vector<ValueT>            d(graph.nodes);
+    std::vector<vertex_descriptor> p(graph.nodes);
+    vertex_descriptor s = vertex(src, g);
+
+    typename property_map<BGraphT, vertex_index_t>::type
+        indexmap = get(vertex_index, g);
+
+    //
+    // Perform Boost reference
+    //
+
+    util::CpuTimer cpu_timer;
+    cpu_timer.Start();
+
+    // TODO: use boost routine to do computation on CPU, e.g.:
+    // dijkstra_shortest_paths(g, s,
+    //        distance_map(boost::make_iterator_property_map(
+    //            d.begin(), get(boost::vertex_index, g))));
+    cpu_timer.Stop();
+    float elapsed = cpu_timer.ElapsedMillis();
+
+    // TODO: extract results on CPU, e.g.:
+    // typedef std::pair<VertexT, ValueT> PairT;
+    // PairT* sort_dist = new PairT[graph.nodes];
+    // typename graph_traits <BGraphT>::vertex_iterator vi, vend;
+    // for (tie(vi, vend) = vertices(g); vi != vend; ++vi)
+    // {
+    //    sort_dist[(*vi)].first  = (*vi);
+    //    sort_dist[(*vi)].second = d[(*vi)];
+    // }
+    // std::stable_sort(
+    //    sort_dist, sort_dist + graph.nodes,
+    //    [](const PairT &a, const PairT &b) -> bool
+    //    {
+    //        return a.first < b.first;
+    //    });
+    // for (VertexT v = 0; v < graph.nodes; ++v)
+    //    distances[v] = sort_dist[v].second;
+    delete[] sort_dist; sort_dist = NULL;
+
+    return elapsed;
+#else
```

Comment:
You don't need boost reference at all for this, the above code starting from line#87 till line#161 is useless.